### DISCUSSION TITLE
[Feature request]: Nur ausgewählte Werte im Inputfeld erlauben

### DISCUSSION THREAD

**Discussion by jdolinga** [SchwarzIT/onyx#3585] [Ideas]
### Description

Als Nutzerin möchte ich (optional) ungültige Eingabewerte in meinem Eingabefeld nicht sehen, damit ich nicht versehentlich eine falsche Eingabe mache und sie nachher wieder korrigieren muss.

### Proposed solution

So lösen wir das Problem aktuell

const handleInputChange = (event: InputEvent) => {
  const newValue = (event.target as HTMLInputElement)!.value;
  if (!props.preventPattern?.test(newValue)) {
    (event.target as HTMLInputElement).value = newValue.slice(0, -1);
  } else {
    emit("inputChange", newValue);
  }
};

### Confirmations

- [X] I have checked that there is not already a feature request for this feature.
- [X] I confirm that are no sensitive information included in this feature request. e.g. credentials, internal project details (for employees of the Schwarz Group) etc.

**Comment by larsrickert** [comment]
Preventing user input is usually not a good idea because it can lead to unexpected behavior.
Instead, I'd recommend to pass the pattern to the `OnyxInput` and set the `showError` property, so errors are shown immediately and not only after the user has left the input. Like this, the user can fix his input directly.

[See playground example](https://playground.onyx.schwarz/?onyxVersion=1.0.0-beta.74#eNp9kkFv2zAMhf+KoNM21NZhO3npsGzooTuswzbsJGxQbcZRJ0uCSDspgvz3UnLSpEXRm8T3LH589E4uY6ynEWQjF9gmG0k44/tLLQm1FAg0xk/a2yGGRGInbvz99trHkcRerFIYhJZoqQpc1vLjmTHB6mThBkXVvg0eSUzGjSAus+mNllq+ZW1WoiGC5FlTf5f1O8XCQs1gjMEXgiE6Q8A3IRadnUTrDCIDZ4aqT7ar+Cky1kPSstjYeOKeqiF04PiDQsFDOnNb7n8K1TAyBpLhKTaW1mLJjuaAxabDKRdxHTYVpBRSjiuVt9TMpRiMTwt1hisv5JxONZhY32HwHPou2/VB4MQbUSq5dsq14duaKGKjVNt5/pYHsFOqPZDycVBH6+dbIMO9kZT1HWzZyfHye3vt99yfkKNZ2f5Z9zYM0TpIN5Esb+EJhXEubL6VWh7x4lhv19D+f6F+hwfgHwkQ0sShPGocag80y1e/vsOWz48ib2V07H5F/AkY3JgZZ9uX0XeMfeYrtNclTOv733i1JfB4HKrsKKdR/OW3/PrK6Cfc9/WHsxT/TZDymxxgXi2S3D8AmUMazQ==)

**Comment by larsrickert** [comment]
I will close this discussion as solved due to inactivity
